### ISSUE TITLE
Mob goal api guide

### ISSUE THREAD

**Issue by masmc05** [PaperMC/docs#265]
It seems that developers see it pretty hard to use the mob goal api, there should be a guide on how to use classes like 
<https://jd.papermc.io/paper/1.20/com/destroystokyo/paper/entity/ai/MobGoals.html>
<https://jd.papermc.io/paper/1.20/com/destroystokyo/paper/entity/ai/Goal.html>

The javadocs aren't really great and there are no tutorials

**Comment by olijeffers0n** [comment]
Yep, this is an API that is often overlooked for NMS so I +1 this guide
